src/dune/install_rules.ml

```diff
@@ -170,17 +178,45 @@ end = struct
                let dune_package_file =
                  Package_paths.dune_package_file ctx pkg
                in
+               let deprecated_meta_and_dune_files =
+                 List.concat_map
+                   ~f:(function
+                     | Deprecated_library_name
+                         { old_public_name =
+                             { deprecated = true
+                             ; public = { name = _, name; sub_dir = None; _ }
+                             }
+                         ; _
+                         } ->
+                       let name = Lib_name.package_name name in
+                       let meta_file =
+                         Package_paths.deprecated_meta_file ctx pkg name
+                       in
+                       let dune_package_file =
+                         Package_paths.deprecated_dune_package_file ctx pkg name
+                       in
+                       ( None
+                       , Install.Entry.make Lib_root meta_file
+                           ~dst:(Package.Name.to_string name ^ "/" ^ Findlib.meta_fn) ) ::
+                       ( None
+                       , Install.Entry.make Lib_root dune_package_file
+                           ~dst:(Package.Name.to_string name ^ "/" ^ Dune_package.fn)) ::
+                       []
+                     | _ -> [])
+                   (Super_context.lib_entries_of_package sctx pkg.name)
```

Comment:
Thinking more about this, we should really use `pkg.deprecated_package_names`. It will be more efficient. The only difference is that we would generate a `META` and `dune-package` file  even if the deprecated packages are empty, but that actually seems like a good thing to me. In particular, this is in line with normal packages.

Reply:
Makes sense, I will change it.

Reply:
This is now done.